File: 4. largest prime factor/python/main.py
from math import sqrt


def largest_prime(n):
    p = 2
    l = []
    while p <= sqrt(n):
        if n % p == 0:
            l.append(p)
            n /= p
        else:
            p += 1
    if n > 1:
        l.append(n)
    l.sort()
    print('Factorization: ', ' * '.join(str(i) for i in l))
    print('Highest prime factor: ', max(l))

File: 4. largest prime factor/cpp/main.cpp
#include <iostream>
#include <math.h>

void prime_factors(long n)
{
    int p = 2;
    while (p <= std::sqrt(n))
    {
        if (n%p == 0)
        {
            std::cout << p << " * ";
            n /= p;
        }
        else
        {
            p++;
        }
    }
    if (n>1) std::cout << n;
}

int main()
{
    prime_factors(600851475143);
}
